README.md
# ConnMatTools - Tools for working with connectivity matrices

## Description

Collects several different methods for analyzing and
working with connectivity matrices in R.  Though primarily oriented
towards marine larval dispersal, many of the methods are general and
useful for terrestrial systems as well.

## Maintainer

[David M. Kaplan](mailto:dmkaplan2000@gmail.com)

## Installation of latest version via github

The latest development version of the package can be installed directly from
[github](https://github.com/dmkaplan2000/ConnMatTools) using the
[devtools](https://cran.r-project.org/package=devtools)
R package. First that package needs to be installed:

     install.packages("devtools")

Once this is done, one can install the development version with:

     require("devtools")
     install_github("ConnMatTools","dmkaplan2000","master")

Replace "master" with another branch, tag or commit to obtain a
different version of the package.

## References

Methods from the following publications are implemented in the package:

Jacobi, M. N., and Jonsson, P. R. 2011. Optimal networks of 
  nature reserves can be found through eigenvalue perturbation theory of the 
  connectivity matrix. Ecological Applications, 21: 1861–1870.
  
Jacobi, M. N., André, C., Döös, K., and Jonsson,
P. R. 2012. Identification of subpopulations from connectivity
matrices. Ecography, 35: 1004-1016.

Grüss, A., Kaplan, D. M., and Lett, C. 2012. Estimating local 
  settler-recruit relationship parameters for complex spatially explicit 
  models. Fisheries Research, 127-128: 34-39.

Kaplan, D. M., Botsford, L. W., and Jorgensen, S. 2006. Dispersal 
  per recruit: An efficient method for assessing sustainability in marine 
  reserve networks. Ecological Applications, 16: 2248-2263.

Kaplan DM, Cuif M, Fauvelot C, Vigliola L, Nguyen-Huu T, Tiavouane J and Lett C 
  (in press) Uncertainty in empirical estimates of marine larval connectivity. 
  ICES Journal of Marine Science. doi:10.1093/icesjms/fsw182.

White, J. W. 2010. Adapting the steepness parameter from 
  stock-recruit curves for use in spatially explicit models. Fisheries 
  Research, 102: 330-334.

Grüss A, Kaplan DM, Hart DR (2011) Relative Impacts of Adult
  Movement, Larval Dispersal and Harvester Movement on the Effectiveness of
  Reserve Networks. PLoS ONE 6:e19960

Beverton RJH, Holt SJ (1957) On the dynamics of exploited fish
  populations. H.M.S.O., London. 533 pp.


NEWS
ConnMatTools 0.3.5
------------------

* Small fixes to correct for multiple class inheritance of matrices in r-devel.

ConnMatTools 0.3.4
------------------

* Small fix to q.rel.conn.dists.func to let you pass extra arguments to the integrate function.

ConnMatTools 0.3.3
------------------

* Add a CITATION file for package.

ConnMatTools 0.3.2
------------------

* Fix for problems due to cumulative distribution function for relative connectivity sometimes
  producing values slightly larger than 1.

ConnMatTools 0.3.1
------------------

* Replacing bootstrap approach to solving finite settler pool problem in relative 
  connectivity code with analytic approach based on hypergeometric distributions.
  
ConnMatTools 0.3.0
------------------

* Finished initial release of package including code for Bayesian-based estimation of 
  relative connectivity.
  
* Also include somes fixes to DPRHomeRange function and a number of minor changes to meet
  new CRAN standards.

ConnMatTools 0.2.1
------------------

* Alpha release of package including Bayesian-based code for estimating relative connectivity.  
  Most of the central functions are available, plus some tests, but a few more pieces 
  need to be added.

ConnMatTools 0.2.0
------------------

* Alpha release of package including initial code for relative connecitvity estimation 
  from field observations

ConnMatTools 0.1.5
------------------

* Small fixes to avoid errors in test script for optimal splitting of connectivity
  matrices.

ConnMatTools 0.1.4
------------------

* Updating version to prepare for submission to CRAN and transfer to github.

ConnMatTools 0.1.2
------------------

* Added function written by Marco Andrello for applying eigenvalue perturbation 
  method in Jacobi & Jonsson (2011) to reserve selection based on connectivity.

ConnMatTools 0.1.1
------------------

* Added several tests, a test dataset and functions for calculating basic statistics 
  on connectivity matrices.

ConnMatTools 0.1
----------------

* Initial release containing code for breaking connectivity matrices into 
  subpopulations and running basic spatial population dynamics models.
* Needs a lot more debugging and test codes.



R/ConnMatTools.R
#' Tools for Working with Connectivity Data
#' 
#' Collects several different methods for analyzing and
#' working with connectivity data in R.  Though primarily oriented
#' towards marine larval dispersal, many of the methods are general
#' and useful for terrestrial systems as well.
#'
#' \tabular{ll}{
#' Package: \tab ConnMatTools\cr
#' Type: \tab Package\cr
#' Version: \tab 0.3.5\cr
#' Date: \tab 2020-02-03\cr
#' License: \tab GPL (>= 2)\cr
#' LazyLoad: \tab no\cr
#' }
#'
#' @author David M. Kaplan \email{dmkaplan2000@@gmail.com}
#' @author Marco Andrello \email{marco.andrello@@gmail.com}
#' 
#' @name ConnMatTools
#' @docType package
#' @title Tools for working with connectivity matrices
#' @keywords package
#' @examples
#' \dontrun{optimalSplitConnMat(CM)}
#' @seealso See \code{\link{optimalSplitConnMat}}, \code{\link{d.rel.conn.beta.prior}}
#' 
#' @references Jacobi, M. N., and Jonsson, P. R. 2011. Optimal networks of 
#'   nature reserves can be found through eigenvalue perturbation theory of the 
#'   connectivity matrix. Ecological Applications, 21: 1861-1870.
#' @references Jacobi, M. N., Andre, C., Doos, K., and Jonsson,
#' P. R. 2012. Identification of subpopulations from connectivity
#' matrices. Ecography, 35: 1004-1016.
#' @references Gruss, A., Kaplan, D. M., and Lett, C. 2012. Estimating local 
#'   settler-recruit relationship parameters for complex spatially explicit 
#'   models. Fisheries Research, 127-128: 34-39.
#' @references Kaplan, D. M., Botsford, L. W., and Jorgensen, S. 2006. Dispersal 
#'   per recruit: An efficient method for assessing sustainability in marine 
#'   reserve networks. Ecological Applications, 16: 2248-2263.
#' @references Kaplan DM, Cuif M, Fauvelot C, Vigliola L, Nguyen-Huu T, Tiavouane J and Lett C 
#'   (in press) Uncertainty in empirical estimates of marine larval connectivity. 
#'   ICES Journal of Marine Science. doi:10.1093/icesjms/fsw182.
#' @references White, J. W. 2010. Adapting the steepness parameter from 
#'   stock-recruit curves for use in spatially explicit models. Fisheries 
#'   Research, 102: 330-334.
#' @references Gruss A, Kaplan DM, Hart DR (2011) Relative Impacts of Adult
#'   Movement, Larval Dispersal and Harvester Movement on the Effectiveness of
#'   Reserve Networks. PLoS ONE 6:e19960
#' @references Beverton RJH, Holt SJ (1957) On the dynamics of exploited fish
#'   populations. H.M.S.O., London. 533 pp.
#'
NULL
